wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceManager.java

```diff
@@ -799,7 +799,7 @@ private Subject sessionUser() {
 
     private String sessionUserNameOr(String nameIfNoUser) {
         final Subject subject;
-        if (EnvironmentContext.getCurrent() != null && (subject = EnvironmentContext.getCurrent().getSubject()) != null) {
+        if ((subject = EnvironmentContext.getCurrent().getSubject()) != Subject.ANONYMOUS) {
```

Comment:
yes I was thinking the same on having isAnonymous() so it delegates the test to Subject and not to all callers